ds-controller-board/driver/driver_test.go

```diff
@@ -174,3 +160,144 @@ func CreateControllerBoardDriver(t *testing.T, virtual bool, initialize bool, ex
 
 	return target
 }
+
+func TestControllerBoardDriver_Initialize(t *testing.T) {
+
+	mocklc := logger.NewMockClient()
+
+	tests := []struct {
+		name      string
+		isVirtual bool
+		config    *device.ServiceConfig
+		lc        logger.LoggingClient
+		wantErr   bool
+	}{
+		{
+			name: "valid virtual case",
+			config: &device.ServiceConfig{
+				DriverConfig: device.Config{
+					VirtualControllerBoard: true,
+				},
+			},
+			lc:      mocklc,
+			wantErr: false,
+		},
+		{
+			name:    "Invalid nil config",
+			config:  nil,
+			lc:      mocklc,
+			wantErr: true,
+		},
+		{
+			name: "Invalid non-virtual case",
+			config: &device.ServiceConfig{
+				DriverConfig: device.Config{
+					VirtualControllerBoard: false,
+				},
+			},
+			lc:      mocklc,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			drv := &ControllerBoardDriver{
+				lc:     tt.lc,
+				config: tt.config,
+			}
+			if err := drv.Initialize(tt.lc, make(chan *dsModels.AsyncValues), make(chan<- []dsModels.DiscoveredDevice)); (err != nil) != tt.wantErr {
+				t.Errorf("ControllerBoardDriver.Initialize() error = %v, wantErr %v", err, tt.wantErr)
+			}
```

Comment:
Too much going on in one line for readablity. Please break it up like we did for your test in other PR and use require.Error/require.NoError